content/news/posts/2024/meet-prestashop9-api.md

```diff
@@ -0,0 +1,386 @@
+
+![PrestaShop 9 API Access Token request using Postman](/assets/images/2024/05/api/postman_access_token.png)
+<p class="text-center text-muted small">Screenshot from Postman where we retrieve access token for our client</p>
+
+{{% notice %}}
+As you can see in the screenshot, I am using project variables in Postman. This way, you can easily switch between different environments (like development, staging, and production) and don't have to change the request data (URL, client_id, etc.) every time you switch environments.
+{{% /notice %}}
+
+After you send the request, you should receive a response with the access token. If you receive information saying "Use HTTPS response", it means that you need to use HTTPS to connect to the API. As mentioned earlier, you can turn off this check in the PrestaShop back office.
+
+If you still have problems with the connection, make sure that the API is enabled in the PrestaShop back office. You can check this in the Advanced Parameters -> Admin API -> Configuration page.
+
+If everything goes well, the response should look similar to this:
+
+```json
+{
+    "token_type": "Bearer",
+    "expires_in": 3600,
+    "access_token": "eyJ0eXAiOiJKV1QiLCJhbGciOiJSUzI1NiJ9.eyJhdWQiOiJ0ZXN0X2NsaWVudCIsImp0aSI6ImQ5OTQzYWQzNDgwNDA3N2QyZGQ5MTBmM2E3YTVmYTdiZTQ1YmMyNzZmM2VmMjA2MGJmZjg2MzhlMzA4ZWE3OWY1ZWI1NzQ0ZmI0N2Y4MDU2IiwiaWF0IjoxNzE2Mjg2MjE5LjQ2NDA3OCwibmJmIjoxNzE2Mjg2MjE5LjQ2NDA4LCJleHAiOjE3MTYyODk4MTkuNDYzMDEsInN1YiI6IiIsInNjb3BlcyI6WyJpc19hdXRoZW50aWNhdGVkIiwiYXBpX2NsaWVudF9yZWFkIiwiYXBpX2NsaWVudF93cml0ZSIsImN1c3RvbWVyX2dyb3VwX3JlYWQiLCJjdXN0b21lcl9ncm91cF93cml0ZSJdfQ.Q6kK0Pl1HVAVrzn5xUrzRO1VSUaw-ygTn9D_rKlfjW3gllUWJiWRaA_pM53RtLId1LkcAfW8nW27CFhQH7TQqLCn4vUPD2t6_s7-3WX_HIqe6MHExib2mW7u_ZXT3bSOyPUOjWIcZsISQR1-noZfOEYDkvgnKDC250zieVqMELgxclMFXKdiLhn83GJnCW35llB1TwAACxdV1uJ_emZGCR3Tsy2IK1pSKPRAb2h-gBre8hqtCmUZ5pdM_L6D_EuUM-aB6iQENiCD6ECmSvqvsqkd3RB73s7PntwniUUafD2GHap1Ttw8pOF7omtT3X0ZLssSX1eQMPDw6JGLFz9caw"
+}
+```
+
+You can now use the access token to authenticate your client in the next requests. You can use it by providing the `Authorization` header with the value `Bearer YOUR_ACCESS_TOKEN`.
+
+Remember that the access token is valid for a limited time. After it expires, you need to get a new one. It is also crucial to provide scopes that your client can use. If you don't give the correct scopes, you won't be able to access some endpoints.
+
+## Information about the API Client
+
+Now that we have the access token, we can get information about the API client. To do that, you need to send a GET request to the `/admin-api/api-client/info` endpoint and provide the access token in the `Authorization` header.
+
+Example request using curl:
+
+```bash
+curl --location 'http://yourdomain.test/admin-api/api-client/infos' \
+--header 'Authorization: Bearer YOUR_ACCESS_TOKEN'
+```
+
+Example request using Postman:
+
+![PrestaShop 9 API Client Info request using Postman](/assets/images/2024/05/api/postman_client_info.png)
```

Comment:
The screenshot doesn't sho the Header part with the authorization header, maybe it can be useful?

Reply:
that's true, I need to explain this because in Postman you can set auth method for the group of requests, so that there's no need to set this header in every single request